src/ert/storage/local_ensemble.py

```diff
@@ -275,67 +276,20 @@ def load_gen_kw_realization(
 
         return np_data, keys
 
-    def load_summary_data_as_df(
-        self, summary_keys: List[str], realizations: List[int]
-    ) -> pd.DataFrame:
-        result = []
-        key_set: Set[str] = set()
-        for realization in realizations:
-            input_path = (
-                self.mount_point / f"realization-{realization}" / "summary-data.nc"
-            )
-            if not input_path.exists():
-                continue
-            with xr.open_dataset(input_path, engine="scipy") as ds_disk:
-                result.append(ds_disk)
-                if not key_set:
-                    key_set = set(sorted(ds_disk["data_key"].values))
-        if not result:
-            raise KeyError(f"Unable to load SUMMARY_DATA for keys: {summary_keys}")
-        df = xr.merge(result).to_dataframe(dim_order=["data_key", "axis"])
-        # realization nr is stored as str in netcdf
-        dropped_keys = key_set - set(summary_keys)
-        if len(dropped_keys) > 0:
-            df.drop(dropped_keys, level="data_key", inplace=True)
-        df.columns = [int(x) for x in df.columns]
-        return df
-
-    def load_gen_data(
-        self, key: str, realizations: List[int]
-    ) -> Tuple[npt.NDArray[np.double], List[int]]:
-        result = []
+    @lru_cache
```

Comment:
Does this yield significant improvements?

Reply:
Yes, quite a bit as long as we are merging multiple observations towards the same response (such as the summary data)

Reply:
Yes, quite a bit as long as we are merging multiple observations towards the same response (such as the summary data)

Reply:
I am not following. AFAIK, caching may help if we call `load_response` many times with the same inputs, i.e., the same `key` and `realizations`.
Are we making many such?

Reply:
I think I get it.
We are calling `ds = source_fs.load_response(group, tuple(ens_active_list))` multiple times in `_get_obs_and_measure_data`...

Reply:
Yes, for `summary` the group will be `summary` regardless of the observation name, so could be quite a few times.